src/am_map.c

```diff
@@ -2319,22 +2319,9 @@ void AM_Drawer (void)
   }
   // [Alaux] Dark automap overlay
   else if (automapoverlay == 2)
-  {
-    int y;
-    byte *dest = screens[0];
-    static int firsttic;
-
-    for (y = 0; y < (SCREENWIDTH << hires) * (SCREENHEIGHT << hires); y++)
-      dest[y] = colormaps[0][viewshade * 256 + dest[y]];
-
-    if (viewshade < 20 && gametic != firsttic)
-    {
-      viewshade += 2;
-      firsttic = gametic;
-    }
-  }
+    overlayshade = V_ShadeScreen(overlayshade, 20);
```

Comment:
I did indeed add an argument to the function to determine just how much the screen must be darkened. In my original implementation, I go for 20 levels instead of 16; it's a little extra contrast that, from my experience, does make a difference in terms of keeping the automap visible.

Reply:
I'd prefer if both cases (automap overlay and menu) would shade by the same value, so we don't have changes in background darkening when the menu gets opened while the automap overlay is up. I'd be fine if you go with `20` for both (okay @rfomin?). Please add `const int targshade = 20;` to `V_ShadeScreen()` so its purpose is as obvious as it is now.

Reply:
Yes, sure, `20` is fine.

Reply:
Well, at first glance, I believe we might still have one problem: while the automap overlay shading affects just the player view, the menu background shading affects not only that, but also the automap, Status Bar and HUD. We could redraw all that before drawing the menu, but I have another idea that I'd like to share: at some point in my implementation, I wanted to make it so that only lighter pixels get darkened, since the objective seems to be to achieve a contrast between elements, and turning already dark pixels to pitch black seems overkill. Unfortunately, I couldn't get it done with what little testing I did. What'd you think?

Reply:
I wouldn't over-engineer this part of the code. The main reason for shading the menu background is that its font graphics do not interfere with the ones from the game view, and it does right that. Similarly, the shaded automap background is meant to still keep the game view slightly visible but keep the HUD and status bar intact, and it does right that.